bodhi/tests/server/services/test_updates.py

```diff
@@ -4548,6 +4548,86 @@ def test_urgent_update_bypass_batched(self, publish, *args):
+
+        build = self.db.query(RpmBuild).filter_by(nvr=nvr).one()
+        build.update.test_gating_status = TestGatingStatus.failed
+        self.assertEqual(build.update.request, UpdateRequest.testing)
+
+        resp = self.app.get('/updates/%s' % nvr,
+                            headers={'Accept': 'text/html'})
+        self.assertIn('text/html', resp.headers['Content-Type'])
+        self.assertIn(nvr, resp)
+        self.assertNotIn(
+            '<div class="alert alert-danger">The update can not be pushed:',
+            resp)
+
+    @mock.patch(**mock_taskotron_results)
+    @mock.patch(**mock_valid_requirements)
+    @mock.patch.dict(config, {'test_gating.required': False})
+    def test_alert_notdisplayed_if_summary_present_but_gating_off(
+            self, publish, *args):
+        """ Ensure that the alert is not being displayed even if there is a
+        summary status (returned normaly by greenwave) set in the database
+        for this update, because the configuration is set otherwise.
+        """
+        nvr = u'bodhi-2.1-1.fc17'
+
+        up_data = self.get_update(nvr)
+        up_data['csrf_token'] = self.app.get('/csrf').json_body['csrf_token']
+        self.app.post_json('/updates/', up_data)
+
+        build = self.db.query(RpmBuild).filter_by(nvr=nvr).one()
+        build.update.test_gating_status = TestGatingStatus.failed
+        build.update.greenwave_summary_string = 'Because Randy said so'
+        self.assertEqual(build.update.request, UpdateRequest.testing)
+
+        resp = self.app.get('/updates/%s' % nvr,
+                            headers={'Accept': 'text/html'})
+        self.assertIn('text/html', resp.headers['Content-Type'])
+        self.assertIn(nvr, resp)
+        self.assertNotIn(
+            '<div class="alert alert-danger">The update can not be pushed: '
+            'Because Randy said so', resp)
+
+    @mock.patch(**mock_taskotron_results)
+    @mock.patch(**mock_valid_requirements)
+    @mock.patch.dict(config, {'test_gating.required': True})
+    def test_alert_displayed_if_summary_present(self, publish, *args):
+        """ Ensure that the alert is displayed if there is a summary status
+        (returned normaly by greenwave) set in the database for this update.
+        """
+        nvr = u'bodhi-2.1-1.fc17'
+
+        up_data = self.get_update(nvr)
+        up_data['csrf_token'] = self.app.get('/csrf').json_body['csrf_token']
+        self.app.post_json('/updates/', up_data)
+
+        build = self.db.query(RpmBuild).filter_by(nvr=nvr).one()
+        build.update.test_gating_status = TestGatingStatus.failed
+        build.update.greenwave_summary_string = 'Because Randy said so'
+        self.assertEqual(build.update.request, UpdateRequest.testing)
+
+        resp = self.app.get('/updates/%s' % nvr,
+                            headers={'Accept': 'text/html'})
+        self.assertIn('text/html', resp.headers['Content-Type'])
+        self.assertIn(nvr, resp)
+        print(resp)
```

Comment:
Let's drop this print statement if you go with the template boolean logic (and thus keep these tests).

Reply:
It is/was there for debugging purposes until I got jenkins to be happy